app/v3/connectors/RetrieveChargeHistoryByChargeReferenceConnector.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v3.connectors
+
+import api.connectors.DownstreamUri.Ifs1Uri
+import api.connectors.httpparsers.StandardDownstreamHttpParser.reads
+import api.connectors.{BaseDownstreamConnector, DownstreamOutcome}
+import config.AppConfig
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import v3.models.request.retrieveChargeHistory.RetrieveChargeHistoryByChargeReferenceRequestData
+import v3.models.response.retrieveChargeHistory.RetrieveChargeHistoryResponse
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RetrieveChargeHistoryByChargeReferenceConnector @Inject() (val http: HttpClient, val appConfig: AppConfig) extends BaseDownstreamConnector {
+
+  def retrieveChargeHistoryByChargeReference(request: RetrieveChargeHistoryByChargeReferenceRequestData)(implicit
+      hc: HeaderCarrier,
+      ec: ExecutionContext,
+      correlationId: String): Future[DownstreamOutcome[RetrieveChargeHistoryResponse]] = {
+
+    val nino            = request.nino.nino
+    val chargeReference = request.chargeReference
+
+    val queryParams = Seq("chargeReference" -> chargeReference.toString)
```

Comment:
probably use ".value" here instead of toString. The toString would only be used to make it behave nicely in s"blah/$chargeReference" strings etc.

Reply:
Also, List( instead of Seq(  :)